specifyweb/frontend/js_src/lib/components/Statistics/index.tsx

```diff
@@ -0,0 +1,599 @@
+              )
+          ),
+        }))
+        .filter(({ items }) => items.length > 0),
+    }));
+  }, [layout, defaultLayout]);
+}
+
+export function StatsPage(): JSX.Element {
+  const [customLayout, setLayout] = usePref(
+    'statistics',
+    'appearance',
+    'layout'
+  );
+  const statsSpec = useStatsSpec();
+
+  const defaultLayout = useDefaultLayout(statsSpec);
+  const layout = customLayout ?? defaultLayout;
+  const [state, setState] = React.useState<
+    | State<
+        'EditingState',
+        {
+          readonly addingItem:
+            | {
+                readonly pageIndex: number;
+                readonly categoryIndex: number;
+              }
+            | undefined;
+        }
+      >
+    | State<'DefaultState'>
+  >({ type: 'DefaultState' });
+  const isEditing = state.type === 'EditingState';
+  const isAddingItem = isEditing && state.addingItem !== undefined;
+  const [activePageIndex, setActivePageIndex] = React.useState<number>(0);
+  const defaultStatsToAdd = useDefaultStatsToAdd(
+    layout[activePageIndex],
+    defaultLayout
+  );
+  console.log(defaultStatsToAdd);
+
+  return (
+    <>
+      {isAddingItem && (
+        <AddStatDialog
+          defaultLayout={defaultStatsToAdd}
+          statsSpec={statsSpec}
+          onAdd={(item): void =>
+            isAddingItem
+              ? setLayout(
+                  replaceItem(layout, activePageIndex, {
+                    ...layout[activePageIndex],
+                    categories: replaceItem(
+                      layout[activePageIndex].categories,
+                      state.addingItem.categoryIndex,
+                      {
+                        ...layout[activePageIndex].categories[
+                          state.addingItem.categoryIndex
+                        ],
+                        items: [
+                          ...layout[activePageIndex].categories[
+                            state.addingItem.categoryIndex
+                          ].items,
+                          item,
+                        ],
+                      }
+                    ),
+                  })
+                )
+              : undefined
+          }
+          onClose={(): void =>
+            setState({
+              type: 'EditingState',
+              addingItem: undefined,
+            })
+          }
+        />
+      )}
+      <div className="h-full w-full bg-[color:var(--form-background)]">
+        <div className="mx-auto flex h-full max-w-[min(100%,var(--form-max-width))] flex-col gap-4 overflow-y-auto  p-4 ">
+          <div className="flex items-center gap-2">
+            <H2 className="text-2xl">{statsText('collectionStatistics')}</H2>
+            <span className="-ml-2 flex-1" />
+            {isEditing && (
+              <Button.Red onClick={(): void => setLayout(defaultLayout)}>
+                {commonText('reset')}
+              </Button.Red>
+            )}
+            <Button.Green
```

Comment:
Split this into two separate buttons, with only one of them being visible at a time